qml/pages/FirstPage.qml

```diff
@@ -41,67 +70,96 @@ Page {
             }
         }
 
-        SilicaListView {
-            id: listView
-            model: []
-            anchors.fill: parent
-            header: PageHeader {
-                title: qsTr("Current Broadcasts")
+        onModelChanged: rowCount = Math.ceil(model.length / columnCount)
+
+        // Cell width and height depend on columnCount, which depends on orientation
+        cellWidth: page.width / columnCount
+        cellHeight: page.width / columnCount / 16.0 * 9.0 + Theme.fontSizeMedium + Theme.paddingMedium
+
+        contentWidth: width
+        contentHeight: gridView.headerItem.height + rowCount * cellHeight + offsetY
+
+        delegate: ListItem {
+            id: listItem
+            width: page.width / columnCount
+            property int baseHeight: gridView.cellHeight
+            height: baseHeight + (index === currIndex ? offsetY : 0.0)
+            contentWidth: width
+            contentHeight: height
+
+            Image {
+                id: programThumbnail
+                y: index >= firstOffsetIndex ? offsetY : 0.0
+                z: 1
+                width: parent.width
+                height: width / 16.0 * 9.0
+                sourceSize.width: imageSizeX
+                sourceSize.height: imageSizeY
+                source: modelData.image && modelData.image.id && modelData.image.available
+                        ? "http://images.cdn.yle.fi/image/upload/w_" + imageSizeX + ",h_" + imageSizeY + ",c_fit/" + modelData.image.id + ".jpg"
+                        : ""
+                BusyIndicator {
+                    anchors.centerIn: parent
+                    size: BusyIndicatorSize.Medium
+                    running: parent.status !== Image.Ready && parent.status !== Image.Error
+                    visible: running
+                }
+                Image {
+                    anchors.centerIn: parent
+                    visible: parent.status === Image.Error
+                    source: "image://theme/icon-l-image"
+                }
+            }
+
+            Label {
```

Comment:
I would like to see some margin around the title, now it's very close to the image. At least some margin before the next image would be nice, I think I also mentioned earlier that margin on both left and right sides the image and title would make the first page follow Sailfish OS app guidelines.